### DISCUSSION TITLE
ERROR: Failed to download minimal PHP for bootstrapping!

### DISCUSSION THREAD

**Discussion by alancwoo** [dokku/dokku#5936] [Q&A] [answered]
I have a php app I'm deploying to dokku with the `heroku-buildpack-php` in my `.buildpacks` file.

When deploying, I receive the error:

```
-----> Warning: Multiple default buildpacks reported the ability to handle this app. The first buildpack in the list below will be used.
       Detected buildpacks: multi php
-----> Multipack app detected
=====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-php
=====> Detected Framework: PHP
remote: -----> Bootstrapping...
remote:
remote:  !     ERROR: Failed to download minimal PHP for bootstrapping!
remote:  !
remote:  !     This is most likely a temporary internal error. If the problem
remote:  !     persists, make sure that you are not running a custom or forked
remote:  !     version of the Heroku PHP buildpack which may need updating.
remote:
remote:  !     Failure during app build
```

Initially I thought maybe the server's packages etc needed to be updated, so I did full release upgrades and updated all packages to latest, and now I have the latest Ubuntu LTS, and versions of `dokku 0.30.7` and `herokuish 0.6.0`

Any help would be greatly appreciated to figure out how to proceed!

**Comment by josegonzalez** [comment]
Can you show the contents of your `.buildpacks` file? Also your `composer.json`.

Finally, where is your server hosted?

**Reply by alancwoo** [reply]
.buildpacks
```
https://github.com/heroku/heroku-buildpack-php
```

composer.json
```
{
  "name": "getkirby/plainkit",
  "description": "Kirby Plainkit",
  "type": "project",
  "keywords": [
    "kirby",
    "cms",
    "starterkit"
  ],
  "homepage": "https://getkirby.com",
  "authors": [
    {
      "name": "Bastian Allgeier",
      "email": "bastian@getkirby.com",
      "homepage": "https://getkirby.com"
    }
  ],
  "support": {
    "email": "support@getkirby.com",
    "issues": "https://github.com/getkirby/starterkit/issues",
    "forum": "https://forum.getkirby.com",
    "source": "https://github.com/getkirby/starterkit"
  },
  "require": {
    "php": "^8.1.0",
    "ext-sqlite3": "*",
    "ext-pdo_sqlite": "*",
    "getkirby/cms": "3.6.2",
    "bnomei/kirby3-autoid": "^2.7",
    "hananils/kirby-colors": "^1.6",
    "bnomei/kirby3-feed": "^1.5",
    "bnomei/kirby3-janitor": "^2.13"
  },
  "config": {
    "optimize-autoloader": true,
    "allow-plugins": {
      "getkirby/composer-installer": true
    }
  }
}
```

Server is hosted on Digital Ocean. In case it's relevant:

/etc/apt/sources.list.d/dokku.list
```
deb https://packagecloud.io/dokku/dokku/ubuntu/ jammy main
```

Nothing buildpack related is set in the app's config. Also, in case it's at all helpful, here are the preceding lines with `trace:on`:

```
remote: + read -r line
remote: + cd /tmp/build
remote: + unprivileged /tmp/buildpacks/00_buildpack-multi/bin/compile /tmp/build /cache /tmp/env
remote: + setuidgid herokuishuser /tmp/buildpacks/00_buildpack-multi/bin/compile /tmp/build /cache /tmp/env
remote: + [[ =====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-php == --* ]]
remote: + [[ =====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-php == ==* ]]
=====>' Downloading Buildpack: https://github.com/heroku/heroku-buildpack-php
=====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-php
remote: + read -r line
remote: + [[ =====> Detected Framework: PHP == --* ]]
remote: + [[ =====> Detected Framework: PHP == ==* ]]
=====>' Detected Framework: PHP
=====> Detected Framework: PHP
remote: + read -r line
remote: -----> Bootstrapping...
remote:
remote:  !     ERROR: Failed to download minimal PHP for bootstrapping!
remote:  !
```

**Reply by josegonzalez** [reply]
Yeah tahts pretty strange, the only time I see this error is if someone is running an older version of herokuish - I'm unable to replicate on my end.

What is the output of `dokku buildpacks:report $APP`? I wonder if you're using an older version of herokuish by accident.

**Reply by alancwoo** [reply]
The output is:

```
=====> cms buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest-20
       Buildpacks global stack:
       Buildpacks list:
       Buildpacks stack:
```

and with `trace:on`, everything I see in a `ps:rebuild` refers to `herokuish:latest-20`

**Reply by josegonzalez** [reply]
Mind restarting docker and retrying?

**Reply by alancwoo** [reply]
I did a `systemctl restart docker` and then `dokku ps:rebuild $APP` with the same issue :(

**Reply by josegonzalez** [reply]
Mind jumping on our slack/discord so I can help you live? This seems like something thats harder to debug async, and I might need server access.

**Reply by alancwoo** [reply]
@josegonzalez sure definitely, will ping you there. Thanks for the help.

**Comment by alancwoo** [comment, marked as answer]
Solved after conversation with @josegonzalez - issue was an outdated `dokku-apt` plugin. `dokku plugin:update apt` worked, as well as having to update `apt-packages` accordingly as `libonig4` no longer available as per https://github.com/dokku/dokku/discussions/4977

Thank you endlessly @josegonzalez for the help, support, and a profoundly valuable oss tool.
